lib/elixir/lib/enum.ex

```diff
@@ -3874,7 +3874,14 @@ defimpl Enumerable, for: Function do
   def member?(_function, _value), do: {:error, __MODULE__}
   def slice(_function), do: {:error, __MODULE__}
 
-  def reduce(function, acc, fun) when is_function(function, 2), do: function.(acc, fun)
+  def reduce(function, acc, fun) when is_function(function, 2) do
```

Comment:
```suggestion
  def reduce(function, acc, fun_element) when is_function(function, 2) do
```

It would be good to rename `fun` to something that can differentiate from the `function` argument